src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/ExternalSettingsProviderService.cs

```diff
@@ -0,0 +1,229 @@
+                    {
+                        if (value is bool boolValue)
+                        {
+                            PackageRestoreConsent.IsAutomatic = boolValue;
+                            return Task.FromResult((ExternalSettingOperationResult)ExternalSettingOperationResult.Success.Instance);
+                        }
+                        break;
+                    }
+                case MonikerSkipBindingRedirects:
+                    {
+                        if (value is bool boolValue)
+                        {
+                            BindingRedirectBehavior.IsSkipped = boolValue;
+                            return Task.FromResult((ExternalSettingOperationResult)ExternalSettingOperationResult.Success.Instance);
+                        }
+                        break;
+                    }
+                case MonikerDefaultPackageManagementFormat:
+                    {
+                        if (value is string strValue)
+                        {
+                            PackageManagementFormat.SelectedPackageManagementFormat = strValue switch
+                            {
+                                MonikerPackagesConfig => 0,
+                                MonikerPackageReference => 1,
+                                _ => throw new ApplicationException("Error saving setting!"),
+                            };
+
+                            PackageManagementFormat.ApplyChanges();
+
+                            return Task.FromResult((ExternalSettingOperationResult)ExternalSettingOperationResult.Success.Instance);
+                        }
+                        break;
+                    }
+                case MonikerShowPackageManagementChooser:
+                    {
+                        if (value is bool boolValue)
+                        {
+                            PackageManagementFormat.Enabled = boolValue;
+                            PackageManagementFormat.ApplyChanges();
+                            return Task.FromResult((ExternalSettingOperationResult)ExternalSettingOperationResult.Success.Instance);
+                        }
+                        break;
+                    }
+                default: break;
+            }
+
+            throw new ApplicationException("Unknown setting!");
+        }
+
+        private static Task<ExternalSettingOperationResult<T>> ConvertValueOrThrow<T>(object input) where T : notnull
+        {
+            if (input is T value)
+            {
+                return Task.FromResult(ExternalSettingOperationResult.SuccessResult(value));
+            }
+
+            throw new ApplicationException("Error reading setting!");
+        }
+
+        private static Task<ExternalSettingOperationResult<T>> ConvertDefaultPackageManagementFormatKeyOrThrow<T>(int input)
+        {
+            if (typeof(T) != typeof(string))
+            {
+                throw new ApplicationException("Error reading setting!");
```

Comment:
Should the error say what the actual type is, so we have more information without needing to debug?

Reply:
I've added a comment and am now just throwing. In summary, this is a programming error, not a runtime error.